src/main/java/carpet/script/CarpetExpression.java

```diff
@@ -2060,24 +2060,24 @@ else if ("!spectating".equalsIgnoreCase(playerName))
             return (_c, _t ) -> retval;
         });
 
-        this.expr.addLazyFunction("entity_area", 7, (c, t, lv) -> {
-            BlockPos center = new BlockPos(
-                    NumericValue.asNumber(lv.get(1).evalValue(c)).getDouble(),
-                    NumericValue.asNumber(lv.get(2).evalValue(c)).getDouble(),
-                    NumericValue.asNumber(lv.get(3).evalValue(c)).getDouble()
-            );
-            Box area = new Box(center).expand(
-                    NumericValue.asNumber(lv.get(4).evalValue(c)).getDouble(),
-                    NumericValue.asNumber(lv.get(5).evalValue(c)).getDouble(),
-                    NumericValue.asNumber(lv.get(6).evalValue(c)).getDouble()
+        this.expr.addLazyFunction("entity_area", -1, (c, t, lv) -> {
+            CarpetContext cc = (CarpetContext) c;
+            BlockArgument centerLocator = BlockArgument.findIn(cc, lv, 1);
+            BlockPos startpos = centerLocator.block.getPos();
+            BlockPos corner = BlockArgument.findIn(cc, lv, centerLocator.offset).block.getPos();
+
+            Box area = new Box(startpos).expand(
+                    corner.getX(),
+                    corner.getY(),
+                    corner.getZ()
             );
             String who = lv.get(0).evalValue(c).getString();
```

Comment:
Also make sure since the number of arguments is not guaranteed, and the error will not be thrown, make sure that there it at least 2 arguments.

Reply:
true